src/TLuaInterpreter.cpp

```diff
@@ -11955,6 +11955,55 @@ int TLuaInterpreter::sendIrc( lua_State * L )
     return 0;
 }
 
+int TLuaInterpreter::setServerEncoding(lua_State * L)
+{
+    Host * pHost = TLuaInterpreter::luaInterpreterMap[L];
+    if (! pHost) {
+        lua_pushstring(L, "setServerEncoding: NULL Host pointer - something is wrong!");
+        return lua_error(L);
+    }
+
+    QString newEncoding;
+    if (! lua_isstring(L, 1)) {
+        lua_pushfstring(L, "setServerEncoding: bad argument #1 type (newEncoding as string expected, got %s!)",
+                        luaL_typename(L, 1));
+        return lua_error( L );
+    }
+    else {
+        newEncoding = QString::fromUtf8(lua_tostring(L,1));
+    }
+
+    QPair<bool, QString> results = pHost->mTelnet.setEncoding(newEncoding);
+
+    if(results.first) {
+        lua_pushboolean(L, true);
+        lua_pushfstring(L, "setServerEncoding: success %s!",
+                        results.second.toLatin1().constData());
+    }
+    else {
+        lua_pushnil(L);
+        lua_pushfstring(L, "setServerEncoding: bad argument #1 value %s",
+                        results.second.toLatin1().constData());
+    }
+    return 2;
+}
+
+int TLuaInterpreter::getServerEncoding(lua_State * L)
```

Comment:
Need a function to get the list of available encodings as well. I think it should be an `encoding = true` map so people can easily check if a certain encoding is available on users Mudlet - @keneanung thoughts on indexed vs map?